Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -26,107 +27,107 @@ public Task<IEnumerable<TServiceModel>> GetLatestSubmissions<TServiceModel>(int
             .ToEnumerableAsync();
 
     public Submission? GetBestForParticipantByProblem(int participantId, int problemId) =>
-            this.GetAllByProblemAndParticipant(problemId, participantId)
-                .Where(s => s.Processed)
-                .OrderByDescending(s => s.Points)
-                .ThenByDescending(s => s.Id)
-                .FirstOrDefault();
+        this.GetAllByProblemAndParticipant(problemId, participantId)
+            .Where(s => s.Processed)
+            .OrderByDescending(s => s.Points)
+            .ThenByDescending(s => s.Id)
+            .FirstOrDefault();
+
+    public IQueryable<Submission> GetByIdQuery(int id) =>
+        this.DbSet
+            .Where(s => s.Id == id);
+
+    public IQueryable<Submission> GetAllByProblem(int problemId)
+        => base.DbSet.Where(s => s.ProblemId == problemId);
+
+    public IQueryable<Submission> GetAllByProblemAndParticipant(int problemId, int participantId) =>
```

Comment:
Do not return the data model. Make a `ServideModel`.
Same for the other methods.

Reply:
Same as above. These are old methods.